### DISCUSSION TITLE
How to run it locally for development?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [danny-avila/LibreChat#1748] [Help Wanted]
Hi,

do you have any guides how to run it locally for development?

What I tried:

## Run backend

I use `npm run backend:dev`. It starts, but weirdly uses the port 27018 for MongoDB, not the default 27017. Anyway, I changed and was opening the browser and got the following error message:

```
Error: ENOENT: no such file or directory, stat '..\librechat\client\dist\index.html'
```

It does not seem to call the vite dev server when running in dev mode, perhaps this would be sorted out automatically if I would run the frontend.#1747 

## Run Frontend

I use `npm run frontend:dev`. Now I get errors like this

```
 The plugin "vite:dep-scan" was triggered by this import

    src/components/Conversations/NewDeleteButton.tsx:2:46:
      2 │ ...t { useDeleteConversationMutation } from 'librechat-data-provider';
        ╵                                             ~~~~~~~~~~~~~~~~~~~~~~~~~


    at failureErrorWithLog (D:\cc\librechat\node_modules\esbuild\lib\main.js:1649:15)
    at D:\cc\librechat\node_modules\esbuild\lib\main.js:1058:25
    at runOnEndCallbacks (D:\cc\librechat\node_modules\esbuild\lib\main.js:1484:45)
 ```
 
 I have never used local dependencies and so far I cannot figure out how it might work.

**Comment by danny-avila** [comment]
Sure I’m not home so I can’t give an in-depth guide at the moment but I will give brief guidance. But you are just missing a step. You need to build the data-provider `npm run build:data-provider` before running the frontend:dev command

**Reply by SebastianStehle** [reply]
Perfect, that works.

**Reply by usmanalisons** [reply]
How does this work without running `npm run frontend` first? In the `server/index.js` file, there are these two lines:

```
const indexPath = path.join(app.locals.paths.dist, 'index.html');
const indexHTML = fs.readFileSync(indexPath, 'utf8');
```

This means the server will look for `dist/index.html` regardless of the environment.

**Comment by melodyxpot** [comment]
You should run `npm run frontend` to build frontend first.
Then you can successfully run `npm run frontend`
